src/sentry/seer/anomaly_detection/store_data.py

```diff
@@ -49,26 +50,35 @@ def format_historical_data(data: SnubaTSResult) -> list[TimeSeriesPoint]:
     return formatted_data
 
 
-def send_historical_data_to_seer(alert_rule: AlertRule, project: Project) -> BaseHTTPResponse:
+def _get_start_and_end_indices(data: SnubaTSResult) -> tuple[int, int]:
     """
-    Get 28 days of historical data and pass it to Seer to be used for prediction anomalies on the alert
+    Helper to return the first and last data points that have event counts.
+    Used to determine whether we have at least a week's worth of data.
+    """
+    start, end = -1, -1
+    indices_with_results = []
+    for i, datum in enumerate(data.data.get("data", [])):
+        if "count" in datum:
+            indices_with_results.append(i)
+    if not indices_with_results:
+        return start, end
+    else:
+        start = indices_with_results[0]
+        end = indices_with_results[-1]
+        assert start <= end
+        return start, end
+
+
+def send_historical_data_to_seer(alert_rule: AlertRule, project: Project) -> AlertRuleStatus:
+    """
+    Get 28 days of historical data and pass it to Seer to be used for prediction anomalies on the alert.
     """
-    base_error_response = BaseHTTPResponse(
-        status=status.HTTP_400_BAD_REQUEST,
-        reason="Something went wrong!",
-        version=0,
-        version_string="HTTP/?",
-        decode_content=True,
-        request_url=SEER_ANOMALY_DETECTION_STORE_DATA_URL,
-    )
-
     snuba_query = SnubaQuery.objects.get(id=alert_rule.snuba_query_id)
     window_min = int(snuba_query.time_window / 60)
     historical_data = fetch_historical_data(alert_rule, snuba_query, project)
 
     if not historical_data:
-        base_error_response.reason = "No historical data available. Cannot create alert_rule."
-        return base_error_response
+        raise ValidationError("No historical data available. Cannot create alert_rule.")
```

Comment:
nit: Since we're not actually modifying the alert_rule here, lets not tie this msg explicitly to the alert_rule
I think just `No historical data available` is good enough for a msg?